core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -557,7 +561,8 @@ private[history] class FsHistoryProvider(conf: SparkConf, clock: Clock)
    */
   private def replay(
       eventLog: FileStatus,
-      bus: ReplayListenerBus): Option[FsApplicationAttemptInfo] = {
+      bus: ReplayListenerBus,
+      eventsFilter: (String) => Boolean = (s: String) => true): Option[FsApplicationAttemptInfo] = {
```

Comment:
Did you mean this way ?

```
  private def replay(
      eventLog: FileStatus,
      bus: ReplayListenerBus,
      eventsFilter: (String) => Boolean): Option[FsApplicationAttemptInfo] = {
```

and have both the places that we call into it provide a filter:

```
val appAttemptInfo = replay(fs.getFileStatus(new Path(logDir, attempt.logPath)),
            replayBus, (s)=> true)
```

and 

```
        val res = replay(fileStatus, bus, (eventString => eventString.startsWith(AppStartEvent)
          || eventString.startsWith(AppEndEvent)))
```